### PR TITLE
Charity 23 manage charity

### PR DESCRIPTION
**Pull request by paivazov**
Adds Charity organisation and managing endpoints (post, put, delete) to it

**References** 
 - #23

### CODE REVIEW

File: api_server/charity/utils/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from common.schemas.responses import ResponseBaseSchema
+
+
+class OrganisationHTTPException(HTTPException):
+    """Custom Organisation not exception."""
+    pass
+
+
+def organisation_exception_handler(request: Request, exc: OrganisationHTTPException):
+    """Handler for UserNotFoundError exception that makes http response.
```
Comment (BorodaUA) [inline api_server/charity/utils/exceptions.py]: Looks like error name spelled wrong.

File: api_server/charity/tests/test_charity.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        assert organisation.users[0].username == "test_john"
+
+    @pytest.mark.asyncio
+    async def test_edit_charity_not_authenticated_should_be_denied(self,
+                                                                   app: FastAPI,
+                                                                   client: AsyncClient,
+                                                                   charity: CharityOrganisation
+                                                                   ):
+        """
+        Test PUT '/charities/{org_id}' endpoint with NOT authenticated user.
```
Comment (BorodaUA) [inline api_server/charity/tests/test_charity.py:92]: Few docstrings here and below looks incomplete.

**Review by BorodaUA — CHANGES_REQUESTED** [review]
Looks like we have here:
```
POST /charities
PUT /charities/<id>
DELETE /charities/<id>
```
But what about
```
GET /charities
GET /charities/<id>
```
How frontend will list a page with all charities, and visit specific charity org page?
Looks a bit weird that not all "CRUD" endpoints created all at once.

**Comment by paivazov** [comment]
> Looks like we have here:
> 
> ```
> POST /charities
> PUT /charities/<id>
> DELETE /charities/<id>
> ```
> 
> But what about
> 
> ```
> GET /charities
> GET /charities/<id>
> ```
> 
> How frontend will list a page with all charities, and visit specific charity org page? Looks a bit weird that not all "CRUD" endpoints created all at once.

It will be in the next pull request. I'm currently working on it on another branch

**Comment by BorodaUA** [comment]
I see, than docstrings fixes and it gets approved from me.

File: api_server/charity/utils/exceptions.py (hunk 1 above)
Comment (paivazov) [reply api_server/charity/utils/exceptions.py]: Thanks, I'll change it

File: api_server/charity/tests/test_charity.py (hunk 2 above)
Comment (paivazov) [reply api_server/charity/tests/test_charity.py:92]: I've written more comprehensive description on this docstring

**Comment by paivazov** [comment]
> I see, than docstrings fixes and it gets approved from me.

Done

File: api_server/charity/utils/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+def organisation_exception_handler(request: Request, exc: OrganisationHTTPException):
+    """Handler for OrganisationHTTPException exception that makes http response.
+
+    Args:
+        request: FastAPI Request object.
+        exc: raised UserNotFoundError.
+
+    Returns:
+    http response for raised UserNotFoundError.
+    """
```
Comment (BorodaUA) [inline api_server/charity/utils/exceptions.py]: I meant those also.

**Review by BorodaUA — CHANGES_REQUESTED** [review]

File: api_server/charity/utils/exceptions.py (hunk 3 above)
Comment (paivazov) [reply api_server/charity/utils/exceptions.py]: Sorry, I must be more attentive. I've changed it

File: api_server/charity/utils/exceptions.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    """Custom Organisation not exception."""
+    pass
+
+
+def organisation_exception_handler(request: Request, exc: OrganisationHTTPException):
+    """Handler for OrganisationHTTPException exception that makes http response.
+
+    Args:
+        request: FastAPI Request object.
+        exc: raised UserNotFoundError.
```
Comment (BorodaUA) [inline api_server/charity/utils/exceptions.py]: The last one I guess...

**Review by BorodaUA — CHANGES_REQUESTED** [review]

File: api_server/charity/utils/exceptions.py (hunk 4 above)
Comment (paivazov) [reply api_server/charity/utils/exceptions.py]: Done

**Review by BorodaUA — APPROVED** [review]

**Review by MaksymShvets — APPROVED** [review]

**Pull request merged**